internal/pkg/cli/deploy/static_site.go

```diff
@@ -166,14 +179,26 @@ func (d *staticSiteDeployer) stackConfiguration(in *StackRuntimeConfiguration) (
 	return conf, nil
 }
 
-func (d *staticSiteDeployer) validateSources() error {
-	for _, upload := range d.staticSiteMft.FileUploads {
-		_, err := d.fs.Stat(upload.Source)
+// convertSources transforms the source's path relative to the project root into the full path.
+func (d *staticSiteDeployer) convertSources() ([]manifest.FileUpload, error) {
+	var convertedFileUploads = make([]manifest.FileUpload, len(d.staticSiteMft.FileUploads))
+	for i, source := range d.staticSiteMft.FileUploads {
+		root := d.rootGetter.ProjectRoot()
+		fullPath := filepath.Join(root, source.Source)
+		fileInfo, err := d.fs.Stat(fullPath)
 		if err != nil {
-			return fmt.Errorf("source %q must be a valid path: %w", upload.Source, err)
+			return nil, fmt.Errorf("source %q must be a valid path relative to the workspace root %q: %w", source.Source, root, err)
 		}
+		convertedFileUploads[i] = manifest.FileUpload{
+			Source:      fullPath,
+			Destination: source.Destination,
+			Recursive:   fileInfo.IsDir(),
```

Comment:
shouldn't this just match the manifest configuration? so:
```suggestion
			Recursive:   source.Recursive,
```